explorer.go

```diff
@@ -139,19 +176,62 @@ func (s APIEndpoint) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	resp, err := c.Get(skycoinURL)
+	var cachedResponse []byte
+	var cachedResponseFound bool
 	if err != nil {
-		msg := "Request to skycoin node failed"
-		log.Println("ERROR:", msg, skycoinURL)
-		http.Error(w, msg, http.StatusInternalServerError)
-		return
+		// Cancel the operation, but not if a previous cached value is available
+		cachedData, found := cacheMgr.Get(s.ExplorerPath)
+		cachedResponseFound = found
+
+		if cachedResponseFound {
+			cachedResponse = cachedData.([]byte)
+		} else {
+			msg := "Request to skycoin node failed"
```

Comment:
If the connection fails and there is no data in the cache, the operation is canceled. But if there is data in cache, these data is used as a fallback.